packages/web-components/fast-components/src/radio/radio.ts

```diff
@@ -0,0 +1,211 @@
+
+        this.$emit("change");
+        this.dispatchEvent(new CustomEvent("change", { bubbles: true, composed: true }));
+        this.checked ? this.classList.add("checked") : this.classList.remove("checked");
+        this.checkedAttribute = this.checked;
+
+        if (this.checked) {
+            this.updateOtherGroupRadios();
+        }
+    }
+
+    private updateOtherGroupRadios(): void {
+        if (this.name !== undefined) {
+            const radioGroup = radioGroups.get(this.name);
+            radioGroup?.forEach((radio: FASTRadio) => {
+                if (radio.getAttribute("value") !== this.value) {
+                    radio.removeAttribute("checked");
+                    radio.checked = false;
+                }
+            });
+        }
+    }
+
+    protected proxy: HTMLInputElement = document.createElement("input");
+
+    /**
+     * Tracks whether the "checked" property has been changed.
+     * This is necessary to provide consistent behavior with
+     * normal input radios
+     */
+    // TODO: marjon check if this is needed for radio buttons?
+    private dirtyChecked: boolean = false;
+
+    constructor() {
+        super();
+        this.proxy.setAttribute("type", "radio");
+        this.addEventListener("keydown", this.keydownHandler);
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.updateForm();
+        if (this.name !== undefined) {
+            const radioGroup = radioGroups.get(this.name);
+            if (radioGroup) {
+                radioGroup.push(this);
+            } else {
+                radioGroups.set(this.name, [this]);
+            }
+        }
+    }
+
+    public disconnectedCallback(): void {
+        if (this.name !== undefined) {
+            const group = radioGroups.get(this.name);
+            if (group) {
+                group.splice(group.indexOf(this), 1);
+            }
+
+            if (group?.length === 0) {
+                radioGroups.delete(this.name);
+            }
+        }
+    }
+
+    private updateForm(): void {
+        const value = this.checked ? this.value : null;
+        this.setFormValue(value, value);
+    }
+
+    public keydownHandler = (e: KeyboardEvent): void => {
+        const group = radioGroups.get(this.name);
+
+        switch (e.keyCode) {
+            case keyCodeArrowRight:
+            case keyCodeArrowUp:
+                if (this.name !== undefined) {
+                    if (group) {
+                        let index = group.indexOf(this) + 1;
+                        while (index < group.length) {
+                            if (!group[index].disabled) {
+                                group[index].checked = true;
+                                group[index].focus();
+                                break;
+                            } else {
+                                index += 1;
+                            }
+                        }
+                    }
+                }
```

Comment:
It seems like this could be extracted out to a separate function to set focus in a specific direction as everything but movement of the index is the same. Essentially having a separate function where you pass a value (+1 or -1) with the direction you want to focus.

Reply:
yep, refactoring now, will do